File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int cnt = 0;

void DFS(int sum, int ind, vector<int> numbers, int target) {
    if(ind == numbers.size()) {
        if(sum == target) {
            cnt++;
        }
        return;
    }
    
    DFS(sum + numbers[ind], ind+1, numbers, target);
    DFS(sum - numbers[ind], ind+1, numbers, target);
}

int solution(vector<int> numbers, int target) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    int answer = 0;
    
    DFS(0, 0, numbers, target);

    answer = cnt;
    
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    int CNT = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(numbers, target, 0, 0);
        answer = CNT;
        return answer;
    }
    
    public void dfs(int[] numbers, int target, int ind, int sum) {
        if(ind+1 > numbers.length) {
            if(sum == target) {
                CNT++;
            }
            return;
        }
        dfs(numbers, target, ind+1, sum+numbers[ind]);
        dfs(numbers, target, ind+1, sum-numbers[ind]);
    }
}